scripts/generate_black_spots_input.py

```diff
@@ -0,0 +1,374 @@
+    """Buffers all intersections"""
+    buffered_intersections = [intersection.buffer(intersection_buffer_meters)
+                              for intersection in get_intersections(roads)]
+    # If intersection buffers overlap, union them to treat them as one
+    return unary_union(buffered_intersections)
+
+
+def split_line(line, max_line_meters):
+    """Checks the line's length and splits in half if larger than the configured max"""
+    if line.length <= max_line_meters:
+        return [line]
+
+    half_length = line.length / 2
+    coords = list(line.coords)
+    for idx, point in enumerate(coords):
+        proj_dist = line.project(Point(point))
+        if proj_dist == half_length:
+            return [LineString(coords[:idx + 1]), LineString(coords[idx:])]
+
+        if proj_dist > half_length:
+            mid_point = line.interpolate(half_length)
+            lines = []
+            lines.extend(
+                split_line(LineString(coords[:idx] + [(mid_point.x, mid_point.y)]), max_line_meters)
+            )
+            lines.extend(
+                split_line(LineString([(mid_point.x, mid_point.y)] + coords[idx:]), max_line_meters)
+            )
+            return lines
+
+
+def get_intersection_parts(roads, int_buffers, max_line_meters):
+    """Finds all segments that intersect the buffers, and all that don't"""
+
+    # Create a spatial index for intersection buffers to efficiently find intersecting segments
+    int_buffers_index = rtree.index.Index()
+    for idx, intersection_buffer in enumerate(int_buffers):
+        int_buffers_index.insert(idx, intersection_buffer.bounds)
+
+    segments_map = {}
+    non_int_lines = []
+    for road in roads:
+        road_int_buffers = []
+        for idx in int_buffers_index.intersection(road.bounds):
+            int_buffer = int_buffers[idx]
+            if int_buffer.intersects(road):
+                if idx not in segments_map:
+                    segments_map[idx] = []
+                segments_map[idx].append(int_buffer.intersection(road))
+                road_int_buffers.append(int_buffer)
+
+        # Collect the non-intersecting segments
+        if len(road_int_buffers) > 0:
+            diff = road.difference(unary_union(road_int_buffers))
+            if 'LineString' == diff.type:
+                non_int_lines.append(diff)
+            elif 'MultiLineString' == diff.type:
+                non_int_lines.extend([line for line in diff])
+        else:
+            non_int_lines.append(road)
+
+    # Union all lines found within a buffer, treating them as a single unit
+    int_multilines = [unary_union(lines) for _, lines in segments_map.items()]
+
+    # Split any long non-intersecting segments. It's not important that they
+    # be equal lengths, just that none of them are exceptionally long.
+    split_non_int_lines = []
+    for line in non_int_lines:
+        splits = split_line(line, max_line_meters)
```

Comment:
Since `split_line` is guaranteed to return an array of lines, you should be able to just do `split_non_int_lines.extend(split_line(line, max_line_meters))` (and you can apparently avoid the for-loop entirely with a nested list comprehension but those are usually confusing so I'm not sure that'd actually be an improvement).